data/src/pixel.rs

```diff
@@ -1,691 +1,292 @@
-    pub fn new(
-        model: ColorModel,
-        components: &[Chromaton],
-        elem_size: u8,
-        be: bool,
-        alpha: bool,
-        palette: bool,
-    ) -> Self {
-        let mut c: [Option<Chromaton>; 5] = [None; 5];
-
-        if components.len() > 5 {
-            panic!("too many components");
-        }
-
-        for (i, v) in components.iter().enumerate() {
-            c[i] = Some(*v);
-        }
-
-        Formaton {
+impl Pixel {
+    /// Constructs a new `Pixe` instance.
+    pub const fn new(model: ColorModel) -> Self {
+        Self {
             model,
 
             primaries: ColorPrimaries::Unspecified,
             xfer: TransferCharacteristic::Unspecified,
             matrix: MatrixCoefficients::Unspecified,
             chroma_location: ChromaLocation::Unspecified,
 
-            components: components.len() as u8,
-            comp_info: c,
-            elem_size,
-            be,
-            alpha,
-            palette,
+            pixel_chroma_info: [None, None, None, None, None],
+
+            pixel_size: 0,
+            is_big_endian: false,
+            has_alpha_plane: false,
+            is_paletted: false,
         }
     }
 
-    /// Returns current color model.
-    pub fn get_model(&self) -> ColorModel {
-        self.model
+    /// Sets colorspace information such as color models, primaries,
+    /// color space conversion functions, and chroma location.
+    pub const fn set_colorspace_info(
+        mut self,
+        primaries: ColorPrimaries,
+        xfer: TransferCharacteristic,
+        matrix: MatrixCoefficients,
+        chroma_location: ChromaLocation,
+    ) -> Self {
+        self.primaries = primaries;
+        self.xfer = xfer;
+        self.matrix = matrix;
+        self.chroma_location = chroma_location;
+        self
     }
 
-    /// Returns current image primaries.
-    pub fn get_primaries(&self) -> ColorPrimaries {
-        self.primaries
+    /// Sets pixel size in bytes for packed formats.
+    pub const fn size(mut self, pixel_size: u8) -> Self {
+        self.pixel_size = pixel_size;
+        self
     }
 
-    /// Returns the total amount of bits needed for components.
-    pub fn get_total_depth(&self) -> u8 {
-        let mut depth = 0;
-        for chromaton in self.comp_info.iter().flatten() {
-            depth += chromaton.depth;
-        }
-        depth
+    /// Pixel data is stored as big-endian.
+    pub const fn is_big_endian(mut self) -> Self {
+        self.is_big_endian = true;
+        self
+    }
+
+    /// Pixel has an alpha plane.
+    pub const fn has_alpha_plane(mut self) -> Self {
+        self.has_alpha_plane = true;
+        self
+    }
+
+    /// Pixel data is paletted.
+    pub const fn is_paletted(mut self) -> Self {
+        self.is_paletted = true;
+        self
+    }
+
+    /// Adds first chromaton pixel plane.
+    pub const fn add_first_plane(mut self, chroma_data: Chromaton) -> Self {
```

Comment:
This is still wrong.

Reply:
Do you have an alternative to propose? Can you explain why it is wrong?